### ISSUE TITLE
AVX support problem

### ISSUE THREAD

**Issue by chester256** [tiny-dnn/tiny-dnn#661]
Recently I am using tiny-dnn to do a image classification project but it performs too slow (4900sec per epoch while 30sec per epoch by keras). So I guess maybe it is caused by avx problem. Then I just add this into my code:

`
  core::backend_t bac = core::backend_t::avx;
  net << convolutional_layer<relu>(64, 64, 3, 1, 8, padding::valid, true, 1, 1, bac)
        << max_pooling_layer<identity>(62, 62, 8, 2)
        << dropout_layer(31 * 31 * 8, 0.2)
        << fully_connected_layer<relu>(31 * 31 * 8, 128, true, bac)
        << fully_connected_layer<softmax>(128, 4, true, bac);
`

Then it was successfully compiled but it came an error when I ran it:

`*** STANFORD C++ LIBRARY 
*** A C++ exception occurred during program execution: 
*** TinyDNN has not been compiled with AVX support.
***
*** Stack trace:
***   tiny_dnn::kernels::fully_connected_op_avx(vector const&, vector const&, vector const&, vector&, tiny_dnn::core::fully_params const&, bool)
***   tiny_dnn::FullyConnectedOp::compute(tiny_dnn::core::OpKernelContext const&)
***   tiny_dnn::fully_connected_layer::forward_propagation(vector const&, vector&)
***   tiny_dnn::layer::forward()
***   tiny_dnn::sequential::forward(vector const&)
***   tiny_dnn::network::fprop(vector const&)
***   void tiny_dnn::network::train_onebatch(tiny_dnn::adagrad&, vector const*, vector const*, int, int, vector const*)
***   void tiny_dnn::network::train_once(tiny_dnn::adagrad&, vector const*, vector const*, int, int, vector const*)
***   construct_cnn(vector&, vector&, vector&, vector&, vector&, vector&)
***   main()`
I have checked that my pc supports avx. Does any one can help? Thank you very much.
My environment: Ubuntu 14.04 + Qt creator

**Comment by beru** [comment]
You need to add `-DCNN_USE_AVX` to `CXXFLAGS` to use AVX backend.
Please make sure your computer has a processor that supports AVX instructions.

**Comment by chester256** [comment]
I have added the line:
`QMAKE_CXXFLAGS += -DCNN_USE_AVX`
into my .pro file, but it still occurred the same problem

**Comment by chester256** [comment]
Now the problem is:
`error: #error "your compiler does not support AVX"`
I am using gcc 4.8.4, is it too old?... Thanks

**Comment by beru** [comment]
Did you specify `-mavx`?

**Comment by chester256** [comment]
Now I specify `-mavx`. It is faster than before.

**Comment by chester256** [comment]
Just one naive question. why the asterisks are displayed vertically when I use the same call back in mnist?
*
*
*
*
*
*
Thank you very much.

**Comment by beru** [comment]
Your guess is as good as mine.
